presto-main/src/main/java/io/prestosql/sql/planner/iterative/rule/InlineProjections.java

```diff
@@ -88,23 +89,29 @@ public Result apply(ProjectNode parent, Captures captures, Context context)
                 .flatMap(entry -> SymbolsExtractor.extractAll(entry).stream())
                 .collect(toSet());
 
-        Assignments.Builder childAssignments = Assignments.builder();
+        Assignments.Builder newChildAssignmentsBuilder = Assignments.builder();
         for (Map.Entry<Symbol, Expression> assignment : child.getAssignments().entrySet()) {
             if (!targets.contains(assignment.getKey())) {
-                childAssignments.put(assignment);
+                newChildAssignmentsBuilder.put(assignment);
             }
         }
         for (Symbol input : inputs) {
-            childAssignments.putIdentity(input);
+            newChildAssignmentsBuilder.putIdentity(input);
+        }
+
+        Assignments newChildAssignments = newChildAssignmentsBuilder.build();
+        PlanNode newChild = child.getSource();
```

Comment:
`PlanNode newChild;` and move the assignment to `else` block.

Also remove `If all assignments end up becoming identity assignments, they'll get pruned by` comment above.